.github/workflows/code.yml

```diff
@@ -89,74 +139,41 @@ jobs:
         id: meta
         uses: docker/metadata-action@v4
         with:
-          images: ghcr.io/${{ github.repository }}
+          images: ${{ env.IMAGE_REPOSITORY }}
           tags: |
-            type=ref,event=branch
             type=ref,event=tag
+            type=raw,value=latest
 
       - name: Set up Docker Buildx
         id: buildx
         uses: docker/setup-buildx-action@v2
 
-      - name: Build developer image for testing
-        uses: docker/build-push-action@v3
-        with:
-          tags: build:latest
-          context: .
-          target: build
-          load: true
-
-      - name: Run tests in the container locked with requirements_dev.txt
-        run: |
-          docker run --name test build bash /project/.github/workflows/container_tests.sh
-          docker cp test:/project/dist .
-          docker cp test:/project/lockfiles .
-          docker cp test:/project/cov.xml .
-
-      - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v3
-        with:
-          name: 3.10-locked/ubuntu-latest
-          files: cov.xml
-
       - name: Build runtime image
         uses: docker/build-push-action@v3
         with:
-          push: ${{ github.event_name != 'pull_request' }}
+          push: ${{ github.event_name == 'push' && startsWith(github.ref, 'refs/tags') }}
+          load: ${{ ! (github.event_name == 'push' && startsWith(github.ref, 'refs/tags')) }}
           tags: ${{ steps.meta.outputs.tags }}
           context: .
-          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha
```

Comment:
I can see this exporting to the cache, but I can't see it importing, not even the base layer...

I might be misunderstanding something about .dockerignore though...

Reply:
You mean you don't see it using the cache in the early build steps in the cache?

I had noticed that but forgot to follow up on it. I'm not entirely sure that the cache-from gha is supposed to persist between workflows (in which case it is useless for our case which does all its docker activity in a single job)

Reply:
according to this https://docs.docker.com/build/building/cache/backends/gha/ I'm doing it right.
this older article discusses how to manually specify cached folders https://evilmartians.com/chronicles/build-images-on-github-actions-with-docker-layer-caching.

Not sure why we don't see a bunch of cache read log entries.